src/scripts/main.js

```diff
@@ -1,3 +1,209 @@
 'use strict';
 
-// write code here
+const body = document.querySelector('body');
+
+let table = document.querySelector('table');
+let headers = [...table.querySelectorAll('th')];
+let tbody = table.querySelector('tbody');
+let rows = [...tbody.querySelectorAll('tr')];
+
+const init = () => {
+  table = document.querySelector('table');
+  headers = [...table.querySelectorAll('th')];
+  tbody = table.querySelector('tbody');
+  rows = [...tbody.querySelectorAll('tr')];
+};
+
+const currencyFormat = (number) =>
+  '$' + new Intl.NumberFormat('en-US').format(number);
+
+// Add active class to row when clicked
+
+rows.forEach((row) => {
+  row.addEventListener('click', () => {
+    rows.forEach((r) => r.classList.remove('active'));
+    row.classList.toggle('active');
+  });
+});
+
+let ASC = true;
+let currentHeader;
+
+// Sort table by column
+headers.forEach((header, i) => {
+  header.addEventListener('click', (e) => {
+    ASC = !ASC;
+
+    if (currentHeader !== i) {
+      ASC = true;
+    }
+    currentHeader = i;
+
+    rows.sort((a, b) => {
+      const cellA = a.querySelectorAll('td')[i].textContent;
+      const cellB = b.querySelectorAll('td')[i].textContent;
+
+      if (Number(numberFormat(cellA))) {
```

Comment:
Here, you are calling `numberFormat` function multiple times for the same values of `cellA` and `cellB`. This could potentially slow down your code if the dataset is large. A more efficient way would be to call this function once for each value and store the result in a variable, then use this variable in your comparisons.